### PR TITLE
promote dataset from fragment

### LINKED ISSUE #16: [Blog Pages] Autoblock Sidebar
The sidebar has the same content for all blog pages in the same geosite (confirm with the customer)
This means that authors should not have to add it to every page. Instead it should be autoblocked on each page with the blog page template, using a fragment.

https://www.aem.live/developer/block-collection/fragment
https://www.aem.live/developer/markup-sections-blocks#auto-blocking

### PR DESCRIPTION
**Pull request by florentin**
Push dataset into the parent section.
Cleanup code.
Better sidebar identification.

Fix #16

Test URLs:

Autoblocked sidebar
- Before: https://main--servicenow--hlxsites.hlx.live/blogs/2023/demo/maximize-efficiency-hyperautomation
- After: https://feature-16-sidebar-eager--servicenow--hlxsites.hlx.live/blogs/2023/demo/maximize-efficiency-hyperautomation

Inline sidebar
- Before: https://main--servicenow--hlxsites.hlx.live/blogs/2023/demo/inline-maximize-efficiency-hyperautomation
- After: https://feature-16-sidebar-eager--servicenow--hlxsites.hlx.live/blogs/2023/demo/inline-maximize-efficiency-hyperautomation

### CODE REVIEW

File: blocks/fragment/fragment.js [hunk 1]
```diff
@@ -38,7 +38,16 @@ export default async function decorate(block) {
...
+      const mainSection = block.closest('.section');
+      if (mainSection) {
+        mainSection.classList.add(...fragmentSection.classList);
+        const dataset = { ...mainSection.dataset, ...fragmentSection.dataset };
+        // block dataset takes precedence so that
+        // data-sectionstatus changes to "loaded" (from "initialized")
+        Object.keys(dataset).forEach((key) => {
+          mainSection.setAttribute(`data-${key}`, dataset[key]);
+        });
+      }
```
Comment (andreituicu) [inline blocks/fragment/fragment.js:50]: This seems to not work as expected.
Screenshot 2023-11-08 at 22 00 14

There are both `data-section-status` and `data-sectionstatus`. 
Same for the extra value: in the fragment it is `data-extra-value` and in the resulting section after loading it is `data-extravalue`.

I believe the main section data set should take precedence over the fragment, as the destination word document configs should be more specific/are overriding what is in the fragment which is more generic. 
Another aspect is: if a section there are multiple blocks and a fragment, the status of the section should not change until all blocks have finished loading, so `data-section-status` should retain the same value after loading the fragment.

File: scripts/scripts.js [hunk 2]
```diff
@@ -164,16 +164,36 @@ function buildArticleSocialShare(main) {
...
+      if (divChildren.length !== 2) {
+        return false;
+      }
+      const isStyle = divChildren[0].textContent.trim().toLowerCase() === key.toLowerCase();
+      const isSidebar = RegExp(`\\b${value}\\b`, 'gi').test(divChildren[1].textContent);
+      return isStyle && isSidebar;
+    },
+  );
+  return targetElement;
+}
```
[the remark this answers is not in the document]
Comment (andreituicu) [reply scripts/scripts.js:230]: Fixed in https://github.com/hlxsites/servicenow/pull/69/commits/cdd585c44a7ef81210f86b67fc7b909dcb773827 and https://github.com/hlxsites/servicenow/pull/69/commits/a0299367c1b62b6d764640341bcfa55002375af7

File: blocks/fragment/fragment.js (hunk 1 above)
Comment (andreituicu) [reply blocks/fragment/fragment.js]: Fixed in https://github.com/hlxsites/servicenow/pull/69/commits/cdd585c44a7ef81210f86b67fc7b909dcb773827 and https://github.com/hlxsites/servicenow/pull/69/commits/a0299367c1b62b6d764640341bcfa55002375af7

File: scripts/scripts.js [hunk 3]
```diff
@@ -213,16 +214,27 @@ function buildArticleSocialShare(main) {
   main.append(div(buildBlock('social-share', { elems: [] })));
 }
 
+function hasInlinedSidebar(main) {
+  const sectionMetas = [...main.querySelectorAll('div.section-metadata')];
```
Comment (sergiucoman) [inline scripts/scripts.js:218]: I assume we go in reverse order because we expect that `sidebar` is at the end.

**Review by sergiucoman — APPROVED** [review]

File: scripts/scripts.js (hunk 3 above)
Comment (andreituicu) [reply scripts/scripts.js:218]: Yes, that was my thinking that the sidebar, by our modeling, is normally one of the last sections, so the main content comes first. I didn't want to risk looking only at the last section.

**Pull request merged**